components/icon/stories/template.js

```diff
@@ -242,42 +166,212 @@ export const Template = ({
 	const classList = {
 		[rootClass]: true,
 		[`spectrum-UIIcon-${iconName}`]: !!(setName === "ui"),
-		[`${rootClass}--${scale}`]: !!(setName === "ui" && scale),
 		[`${rootClass}--size${size?.toUpperCase()}`]: !!(
 			(!setName || setName === "workflow") &&
 			size
 		),
 		...customClasses.reduce((a, c) => ({ ...a, [c]: true }), {}),
 	};
 
-	if (svgString) {
-		const classesAsString = Object.entries(classList).reduce((acc, [key, value]) => {
-			if (value) acc += `${key} `;
-			return acc;
-		}, "");
+	/**
+	 * Display full SVG file markup from global IconLoader data, when not using a reference to the sprite sheet.
+	 */
+	if (!useRef) {
+		let svgString;
+		if (loaded?.icons && loaded?.icons[setName]?.[idKey]) {
+			svgString = loaded.icons[setName][idKey];
+		}
+
+		// Return the individual SVG's entire markup.
+		if (svgString) {
+			const classesAsString = Object.entries(classList).reduce((acc, [key, value]) => {
+				if (value) acc += `${key} `;
+				return acc;
+			}, "");
 
-		return html`${unsafeSVG(
-			svgString.replace(/<svg/, `<svg class="${classesAsString}" focusable="false" aria-hidden="true" role="img"`)
-		)}`;
+			return html`${unsafeSVG(
+				svgString.replace(/<svg/, `<svg class="${classesAsString}" focusable="false" aria-hidden="true" role="img"`)
+			)}`;
+		}
+		else {
+			console.warn(`Could not find SVG markup for "${idKey}" in context.loaded.icons. Did you pass through context? Falling back to using the sprite sheet reference instead.`);
+		}
 	}
 
-	// ui ID: #spectrum-css-icon-${idKey}
-	// workflow ID: #spectrum-icon-(18|24)-${idKey}
-	const iconID =
-		setName !== "workflow"
-			? `spectrum-css-icon-${idKey}`
-			: `spectrum-icon-${scale !== "medium" ? "24" : "18"}-${idKey}`;
+	// ID of the icon within the sprite sheet for its icon set.
+	const iconID = getSpriteSheetName(idKey, setName);
 
+	// Return SVG markup with a reference to the icon ID within the sprite sheet.
 	return html`<svg
 		class=${classMap(classList)}
 		id=${ifDefined(id)}
-		style=${ifDefined(inlineStyle)}
+		style=${ifDefined(fill ? `color: ${fill};` : undefined)}
 		focusable="false"
 		aria-hidden="true"
-		aria-labelledby=${idKey}
+		aria-label=${iconName}
 		role="img"
 	>
 		<title id=${idKey}>${idKey.replace(/([A-Z])/g, " $1").trim()}</title>
 		<use xlink:href="#${iconID}" href="#${iconID}" />
 	</svg>`;
 };
+
+/**
+ * Display all icons in the icon set within a grid.
+ */
+export const FullIconSetTemplate = (args, context) => {
+	return html`
+		<div
+			style=${styleMap({
+				"display": "grid",
+				"grid-template-columns": "repeat(4 , 150px)",
```

Comment:
I wonder if we should let it do its flexbox thing and flow full screen to get more visible at a time? Make it easier to quickly scan it visually maybe?

Reply:
The grid columns for the icon lists will now `auto-fill` 🎉